src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -21,91 +28,66 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
-						.Select(static (combined, cancellationToken) => (UserGeneratedClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right))
-						.Combine(context.CompilationProvider)
-						.Select(static (combined, cancellationToken) => (combined.Left.UserGeneratedClassesProvider, combined.Left.MauiControlsAssemblySymbolProvider, Compilation: combined.Right));
+			static (context, cancellationToken) =>
+			{
+				var compilation = context.SemanticModel.Compilation;
 
-		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
-		Execute(context, collectedValues.Compilation, collectedValues.UserGeneratedClassesProvider, collectedValues.MauiControlsAssemblySymbolProvider));
-	}
+				var iTextStyleInterfaceSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
+				var iAnimatableInterfaceSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
 
-	static void Execute(SourceProductionContext context, Compilation compilation, ImmutableArray<ClassDeclarationSyntax> userGeneratedClassesProvider, IAssemblySymbol mauiControlsAssemblySymbolProvider)
-	{
-		var textStyleSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
-		var iAnimatableSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
-
-		if (textStyleSymbol is null || iAnimatableSymbol is null)
-		{
-			var diag = Diagnostic.Create(TextColorToDiagnostic.MauiReferenceIsMissing, Location.None);
-			context.ReportDiagnostic(diag);
-			return;
-		}
+				if (iTextStyleInterfaceSymbol is null || iAnimatableInterfaceSymbol is null)
+				{
+					throw new Exception("There's no .NET MAUI referenced in the project.");
+				}
 
-		var textStyleClassList = new List<(string ClassName, string ClassAcessModifier, string Namespace, string GenericArguments, string GenericConstraints)>();
+				var classSymbol = (INamedTypeSymbol?)context.SemanticModel.GetDeclaredSymbol(context.Node);
 
-		// Collect Microsoft.Maui.Controls that Implement ITextStyle
-		var mauiTextStyleImplementors = mauiControlsAssemblySymbolProvider.GlobalNamespace.GetNamedTypeSymbols().Where(x => x.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
-				&& x.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default));
+				// If the ClassDlecarationSyntax doesn't implements those interfaces we just return null
+				if (classSymbol is null
+					|| !(classSymbol.AllInterfaces.Contains(iAnimatableInterfaceSymbol, SymbolEqualityComparer.Default)
+							&& classSymbol.AllInterfaces.Contains(iTextStyleInterfaceSymbol, SymbolEqualityComparer.Default)))
+				{
+					return null;
+				}
 
-		foreach (var namedTypeSymbol in mauiTextStyleImplementors)
-		{
-			textStyleClassList.Add((namedTypeSymbol.Name, "internal", namedTypeSymbol.ContainingNamespace.ToDisplayString(), namedTypeSymbol.TypeArguments.GetGenericTypeArgumentsString(), namedTypeSymbol.GetGenericTypeConstraintsAsString()));
-		}
+				return classSymbol;
```

Comment:
Using symbols as part of generator pipeline **must** be avoided. This can break incrementality.